governance/program/src/instruction.rs

```diff
@@ -443,6 +444,14 @@ pub enum GovernanceInstruction {
     ///   4. `[signer]` Payer
     ///   5. `[]` System
     CreateTokenOwnerRecord {},
+
+    /// Updates ProgramMetadata account
+    /// The instruction dumps information implied by the program's code into a persistent account
+    ///
+    ///  0. `[writable]` ProgramMetadata account. PDA seeds: ['metadata']
+    ///  1. `[signer]` Payer
+    ///  2. `[]` System
+    UpdateProgramMetadata {},
```

Comment:
nit: Since it could be confusing between the governance program and the governed program, let's make this totally clear and call it `GovernanceProgramMetadata` everywhere
```suggestion
    UpdateGovernanceProgramMetadata {},
```

Reply:
I renamed it, then thought about it hard and reverted the change :)

The name become too verbose and it was still to close to `GovernanceMetadata` account which I'm planning to add in future versions while `ProgramMetadata` can be used consistently as a pattern accross different programs.

At the same time I want to remove the concrete governance types `ProgramGovernance`, `MintGovernance` and `TokenGovernance`. They are not really needed in the program and they only support UI.

Reply:
Fair enough